test-network-function/lifecycle/suite.go

```diff
@@ -113,58 +113,74 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
-		log.Debugf("Waiting for deployments to get ready, remaining: %d deployments", len(notReadyDeployments))
-		if len(notReadyDeployments) == 0 {
+		_, notReadyPodSets = GetPodSets(namespace, resourceType)
+		log.Debugf("Waiting for PodSets to get ready, remaining: %d PodSets", len(notReadyPodSets))
+		if len(notReadyPodSets) == 0 {
 			break
 		}
 		time.Sleep(pollingPeriod)
 		elapsed += pollingPeriod
 	}
-	return len(notReadyDeployments)
+	return len(notReadyPodSets)
 }
 
 // restoreDeployments is the last attempt to restore the original test deployments' replicaCount
 func restoreDeployments(env *config.TestEnvironment) {
-	for _, deployment := range env.DeploymentsUnderTest {
+	for i := range env.DeploymentsUnderTest {
 		// For each test deployment in the namespace, refresh the current replicas and compare.
-		deployments, notReadyDeployments := getDeployments(deployment.Namespace)
-
-		if len(notReadyDeployments) > 0 {
-			// Wait until the deployment is ready
-			notReady := waitForAllDeploymentsReady(deployment.Namespace, scalingTimeout, scalingPollingPeriod)
-			if notReady != 0 {
-				collectNodeAndPendingPodInfo(deployment.Namespace)
-				log.Fatalf("Could not restore deployment replicaCount for namespace %s.", deployment.Namespace)
-			}
-		}
+		refreshReplicas(&env.DeploymentsUnderTest[i], env)
+	}
+}
+
+// restoreStateFullSet is the last attempt to restore the original test PodSets' replicaCount
+func restoreStateFullSet(env *config.TestEnvironment) {
+	for i := range env.StateFullSetUnderTest {
+		// For each test deployment in the namespace, refresh the current replicas and compare.
+		refreshReplicas(&env.StateFullSetUnderTest[i], env)
+	}
+}
+
+func refreshReplicas(podset *configsections.PodSet, env *config.TestEnvironment) {
+	podsets, notReadyPodsets := GetPodSets(podset.Namespace, podset.Type)
 
-		if deployment.Hpa.HpaName != "" { // it have hpa and need to update the max min
-			runHpaScalingTest(deployment)
+	if len(notReadyPodsets) > 0 {
+		// Wait until the podset is ready
+		notReady := waitForAllPodSetsReady(podset.Namespace, scalingTimeout, scalingPollingPeriod, podset.Type)
+		if notReady != 0 {
+			collectNodeAndPendingPodInfo(podset.Namespace)
+			log.Fatalf("Could not restore %s replicaCount for namespace %s.", string(podset.Type), podset.Namespace)
 		}
-		if deployments[deployment.Name].Replicas != deployment.Replicas {
-			log.Warn("Deployment ", deployment.Name, " replicaCount (", deployment.Replicas, ") needs to be restored.")
+	}
+	if podset.Hpa.HpaName != "" { // it have hpa and need to update the max min
+		runHpaScalingTest(podset, podset.Hpa.MinReplicas, podset.Hpa.MaxReplicas)
+	}
+	key := podset.Namespace + ":" + podset.Name
+	dep, ok := podsets[key]
+	if ok {
+		if dep.Replicas != podset.Replicas {
+			log.Warn("Deployment ", podset.Name, " replicaCount (", podset.Replicas, ") needs to be restored.")
 
 			// Try to scale to the original deployment's replicaCount.
```

Comment:
If statefulsets can also be scaled through hpa, then this text also needs to be updated to deployment/statefulset.